_pytest/skipping.py

```diff
@@ -147,10 +148,17 @@ def getexplanation(self):
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_runtest_setup(item):
-    evalskip = MarkEvaluator(item, 'skipif')
-    if evalskip.istrue():
-        item._evalskip = evalskip
-        pytest.skip(evalskip.getexplanation())
+    eval_skipif = MarkEvaluator(item, 'skipif')
+
+    if eval_skipif.istrue():
+        item._evalskip = eval_skipif
+        pytest.skip(eval_skipif.getexplanation())
+    elif isinstance(item.keywords.get('skip'), MarkInfo):
```

Comment:
Thanks for the work! I noticed two problems while playing around with it however:

* `reason` is not handled when missing, so the user sees this 
  when using a plain `@pytest.mark.skip`:
  
  ```
  SKIP [1] test_xx.py:2: <Skipped instance>
  ```
  
* the user could still use `pytest.mark.skip('False')` and the test would
  not skip.
  

This fixes it, but it is not pretty :sweat_smile::

```python
@pytest.hookimpl(tryfirst=True)
def pytest_runtest_setup(item):
    evalskip = MarkEvaluator(item, 'skip')
    if evalskip.holder is not None:
        evalskip.reason = evalskip.holder.kwargs.get('reason', 'explicit skip')
        evalskip.result = True
        item._evalskip = evalskip
        pytest.skip(evalskip.getexplanation())

    evalskip = MarkEvaluator(item, 'skipif')
    if evalskip.istrue():
        item._evalskip = evalskip
        pytest.skip(evalskip.getexplanation())
    ...
```

The implementations lets the user don't pass any `reason` in which case it will use `explicit skip`... perhaps we want to enforce it?